frontend/javascripts/admin/task/task_search_form.js

```diff
@@ -142,98 +144,89 @@ class TaskSearchForm extends React.Component<Props, State> {
     });
   };
 
-  handleSearchFormSubmit = (isRandom: boolean, event: ?SyntheticInputEvent<*>) => {
-    this.handleFormSubmit(isRandom, this.props.onChange, event);
+  handleSearchFormSubmit = (isRandom: boolean) => {
+    this.handleFormSubmit(isRandom, this.props.onChange);
   };
 
   handleDownloadAllTasks = () => {
     this.handleFormSubmit(false, this.props.onDownloadAllTasks);
   };
 
   handleReset = () => {
-    this.props.form.resetFields();
+    const form = this.formRef.current;
+    if (!form) {
+      return;
+    }
+    form.resetFields();
     this.setState({ fieldValues: {} });
     this.props.onChange({});
   };
 
   render() {
     const { isLoading } = this.props;
-    const { getFieldDecorator } = this.props.form;
     const formItemLayout = {
       labelCol: { span: 5 },
       wrapperCol: { span: 19 },
     };
 
     return (
-      <Form onSubmit={evt => this.handleSearchFormSubmit(false, evt)}>
+      <Form onFinish={() => this.handleSearchFormSubmit(false)} ref={this.formRef}>
```

Comment:
Let's rename all `...FormSubmit` methods to `...FormFinish` to indicate, that the `onFinish` method is used and not manual validation is needed.

Reply:
Makes sense. Are you doing that?

Reply:
I think I already pushed that yesterday.